GetMessages/handler.ts

```diff
@@ -145,78 +98,41 @@ export const GetMessagesHandler = (
         minimumId
       }) =>
         pipe(
-          messageModel.findMessages(fiscalCode, pageSize, maximumId, minimumId),
-          TE.mapLeft(failure => ResponseErrorQuery(failure.kind, failure)),
-          TE.map(i => mapAsyncIterator(i, A.rights)),
-          TE.map(i =>
-            mapAsyncIterator(i, A.filter(RetrievedNotPendingMessage.is))
-          ),
-          TE.map(i => mapAsyncIterator(i, A.map(retrievedMessageToPublic))),
-          TE.chainW(i =>
-            // check whether we should enrich messages or not
-            pipe(
-              TE.fromPredicate(
-                () => shouldEnrichResultData === true,
-                () =>
-                  // if no enrichment is requested we just wrap messages
-                  mapAsyncIterator(
-                    i,
-                    A.map(async e =>
-                      E.right<Error, CreatedMessageWithoutContent>(e)
-                    )
-                  )
-              )(i),
-              TE.map(j =>
-                mapAsyncIterator(j, async m =>
-                  enrichMessagesStatus(context, messageStatusModel)(m)()
-                )
-              ),
-              TE.map(j =>
-                mapAsyncIterator(j, filterMessages(shouldGetArchivedMessages))
-              ),
-              TE.map(j =>
-                mapAsyncIterator(j, x => [
-                  // Do not enrich messages if errors occurred
-                  ...pipe(
-                    A.lefts(x),
-                    A.map(async y => E.left(y))
+          functionSelector.select({ fiscalCode }),
+          getMessagesFunction =>
+            getMessagesFunction({
+              context,
+              fiscalCode,
+              maximumId,
+              minimumId,
+              pageSize,
+              shouldEnrichResultData,
+              shouldGetArchivedMessages
+            }),
+          TE.chainW(paginatedItems =>
+            !shouldEnrichResultData
```

Comment:
Maybe I'm wrong but the semantic of this parameter has been changed. The previous version tell handler to enrich messages with status, content and service. In this version we enrich only services because enrichment happens anyway.

Reply:
Actually right now service enrichment is done only when `shouldEnrichResultData` is `true`

Reply:
So enrichment happens in any case. If it is the case we perform message enrichment (for status and content) also for older app versions. Do we expect this behaviour?

Reply:
No,  serice enrichment happens only when `shouldEnrichResultData` is `true`. For older app we have `false` as default

Reply:
yes I mean status and content are always enriched despite of app version ( regardless to `shouldEnrichResultData`'s value)